compiler/rustc_middle/src/ty/upvar.rs

```diff
@@ -0,0 +1,72 @@
+use crate::ty;
```

Comment:
This is the only file that I'm not sure holds it weight. I feel like this should also include some of the min capture related stuff or just things related to closure captures in general.